### PR TITLE
Add page with living style guide

### PR DESCRIPTION
**Pull request by straight-shoota**
This adds a new page at `/style-guide` which contains examples of many common design components on this website. It serves as a living style guide to help ensure consistency of this site's style. It's easier to monitor relevant changes on a single page than having to browse several pages in order to find instances of less common design components.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -65,3 +67,4 @@ kramdown:
     block:
       css_class: code_section
 
+repository: crystal-lang/crystal-website
```
Comment (beta-ziliani) [inline _config.yml:70]: What is this for? It's unrelated to this PR, right?

File: _style_guide/typography-code-block.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
+title: Code Block
+type: typography
+---
+
+```crystal
```
Comment (beta-ziliani) [inline _style_guide/typography-code-block.md:6]: I suggest to add what the block shows (same for the rest of the code blocks below):

```suggestion
Crystal code:
```crystal
```

File: _style_guide/elements_team-member.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+title: Team Member
+type: elements
+---
+<div class="team-member">
```
Comment (beta-ziliani) [inline _style_guide/elements_team-member.md:5]: The indentation here is broken.

File: _style_guide/typography-list-ordered.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
```
Comment (beta-ziliani) [inline _style_guide/typography-list-ordered.md:1]: The meaning are flipped. This is an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists):

1. They
2. Needs
3. Numbers

This is an [unordered list](https://www.markdownguide.org/basic-syntax/#unordered-lists):

 * No
 * Numbers
 * Here

**Review by beta-ziliani — COMMENTED** [review]
This is AWESOME! 🤩  Left some minor comments.

I wonder if it's possible to also display the `.md` to see where the HTML came from.

File: _config.yml (hunk 1 above)
Comment (straight-shoota) [reply _config.yml:70]: It's used in the `component` include to generate links to the source code.

https://github.com/straight-shoota/crystal-website/blob/b88a3177e52dbb946c01db5033dbdd39f6521457/_includes/style-guide/component.html#L5-L6

**Comment by straight-shoota** [comment]
> I wonder if it's possible to also display the .md to see where the HTML came from.

Yeah, this was my original intention as well. However, it proves to be nontrivial.

The capabilities of Jekyll are a bit limited for this kind of thing. `{% include_relative {{ component.path }} %}` mostly works, but I couldn't get it to remove the frontmatter. And it breaks when the component includes another include (which happens for the *Post Teaser*).
This should be easy to solve with a plugin. I'll look into that, but add it as a separate enhancement.

File: _style_guide/typography-code-block.md (hunk 2 above)
Comment (straight-shoota) [reply _style_guide/typography-code-block.md:6]: I don't think this is necessary. The language should be obvious and is clearly visible in the source code.

Comment (beta-ziliani) [reply _style_guide/typography-code-block.md:6]: That's true now that you show the md, smarty-pants ;-)

**Review by beta-ziliani — APPROVED** [review]
🚀

**Pull request merged**